webgoat-lessons/cross-site-scripting/src/main/java/org/owasp/webgoat/xss/DOMCrossSiteScripting.java

```diff
@@ -40,12 +40,15 @@ public AttackResult completed(@RequestParam Integer param1,
         UserSessionData userSessionData = getUserSessionData();
         SecureRandom number = new SecureRandom();
         userSessionData.setValue("randValue", String.valueOf(number.nextInt()));
-
+        
         if (param1 == 42 && param2 == 24 && request.getHeader("webgoat-requested-by").equals("dom-xss-vuln")) {
-            return success(this).output("phoneHome Response is " + userSessionData.getValue("randValue").toString()).build();
+            if (userSessionData != null && userSessionData.getValue("randValue") != null){
+                return success(this).output("phoneHome Response is " + userSessionData.getValue("randValue").toString()).build();
+            }
         } else {
             return failed(this).build();
         }
```

Comment:
Build complains here with: "missing return statement".  I think this fixes it:
```suggestion
        }
        return failed(this).build();
```